### ISSUE TITLE
Getting unrelated (react/no-this-in-sfc) violation error

### ISSUE THREAD

**Issue by psonawane-servicetitan** [yannickcr/eslint-plugin-react#1960]
I ran the following simple js program. But I got an error at line this.getA(); which is totally unrelated here. The error said "stateless functional components should not use this (react/no-this-in-sfc)"

```
class Demo {
    getA = () => 'a';

    hasBug = () => {
        this.getA();

        return null;
    };
}
```

What did you expect to happen?
No error expected

What actually happened? Please include the actual, raw output from ESLint.
Got this error - stateless functional components should not use this (react/no-this-in-sfc) for line  ```this.getA()```

**Comment by ljharb** [comment]
Component detection is probably recognizing it as an SFC since it returns null; this seems like a case where we should have less confidence because a) it's named not in PascalCase, b) it's a class field.

cc @alexzherdev

**Comment by alexzherdev** [comment]
We're bitten again by the fact that `Components.get` can return something we're not entirely sure is a component (confidence = 1).
I can make a local fix in this rule to make sure this doesn't happen. But making `Components.get` always require confidence of 2 breaks 37 different tests in the suite 😕

**Comment by ljharb** [comment]
@alexzherdev in this case it should be reporting a lower confidence than 1, because it's a class field, and because it's not named in PascalCase.

**Comment by bendtherules** [comment]
Not sure if this is due to the same reason.
Another sample causing error:
```
import React from 'react';

class Extended extends Base {
  getReminderWidgetJSX = () => {
    this.abc();

    return (
      <div>
        Hello World
      </div>
    );
  }
}
```
